Funcky/Extensions/EnumerableExtensions/ForEach.cs

```diff
@@ -16,5 +17,13 @@ public static void ForEach<T>(this IEnumerable<T> elements, Action<T> action)
                 action(element);
             }
         }
+
+        /// <summary>
+        /// The IEnumerable version of foreach. You can apply an Func{T, Unit} to each element. This is only useful when you have side effects.
+        /// </summary>
+        /// <typeparam name="T">the inner type of the enumerable.</typeparam>
+        public static Unit ForEach<T>(this IEnumerable<T> elements, Func<T, Unit> action)
```

Comment:
Do we also need a similar thing for `Option.AndThen`?

Reply:
Option.AndThen has already an overload for Func<T, TResult> where TResult can be Unit.
This should also apply to all Match functions which already support Func<T, TResult> and Action.